UPGRADE.md

```diff
@@ -1939,6 +1939,131 @@ If you've declared custom Record types, you'll need to upgrade your code as foll
 }
 ```
 
+## react-final-form Has Been Replaced By react-hook-form
+
+### `<FormWithRedirect>` Has Been Renamed to `<Form>`
+
+The form components don't handle redirection at all anymore as it happens in side effects, the `<FormWithRedirect>` has been renamed to `<Form>`.
+
+To upgrade simply replace all occurrences of `FormWithRedirect` with  `Form`.
+
+```diff
+- import { FormWithRedirect } from 'react-admin';
++ import { Form } from 'react-admin';
+
+export const MyForm = () => (
+-    <FormWithRedirect
++    <Form
+        render={() => ...}
+    />
+);
+```
+
+### `<FormWithRedirect>` Props Have Changed
+
+`<FormWithRedirect>` used to accept [`react-final-form` `<Form>` props](https://final-form.org/docs/react-final-form/types/FormProps). It now accepts [`react-hook-form` `useForm` props](https://react-hook-form.com/api/useform). This also affects the other form components (`SimpleForm`, `TabbedForm`, etc.)
+
+The most commonly used prop is probably `initialValues` which is now named `defaultValues`:
+
+```diff
+const PostCreate = () => (
+    <Create>
+        <SimpleForm
+-            initialValues={{ title: 'A default title' }}
++            defaultValues={{ title: 'A default title' }}
+        >
+            ...
+        </SimpleForm>
+    </Create>
+)
+```
+
+However, we kept the `validate` function prop which we automatically translate as a custom [`react-hook-form` `resolver`](https://react-hook-form.com/api/useform#validationResolver).
+
+This also means you can now use [`yup`](https://github.com/jquense/yup), [`zod`](https://github.com/colinhacks/zod), [`joi`](https://github.com/sideway/joi), [superstruct](https://github.com/ianstormtaylor/superstruct), [vest](https://github.com/ealush/vest) or any [resolvers](https://react-hook-form.com/api/useform#validationResolver) supported by `react-hook-form` to apply schema validation.
+
+### `<FormWithRedirect>` Render Function Arguments Have Changed
```

Comment:
I thought we got rid of the render prop?

Reply:
Not yet and I'm not sure we actually can